### DISCUSSION TITLE
Will we get any rigging capabilities in friction future updates?

### DISCUSSION THREAD

**Discussion by freewarecamp** [friction2d/friction#77] [Ideas]
Is it possible to get a rigging capability like after effect joysticks? That would be really useful

**Comment by rodlie** [comment]
Don't know about joysticks (:P), but you got null objects that can be used for basic rigging stuff.
![Screenshot from 2023-09-08 09-27-39](https://github.com/friction2d/friction/assets/34516798/bfdfdf66-fcb5-4612-a0a7-64a90a94ff99)

**Comment by freewarecamp** [comment]
In After effects you create 4 keyframes (Example: Up, Down, Left, Right) for the full animation. Then, using the joysticks controller we store the keyframes on it. After that we can make animation using the controller. There is also a slider to control 2 keyframes in joysticks. That's how the rigging works. I hope you got the idea

https://github.com/friction2d/friction/assets/86367188/2e67b2db-02c6-49ee-ac0f-55b5f4f48c66



https://github.com/friction2d/friction/assets/86367188/dbb28c76-50b4-4f0a-997e-18976b0922f3

is it possible to bring feature like this?

**Comment by AlexKiryanov** [comment]
Hi all! I'll try to explain.
The operating principle of a joystick or DUIK is as follows.
There is an element in the form of a shape layer with frames.
With the help of a script, the slider allows you to move from one frame in the shape layer to another. Roughly speaking, the script controls the morphing.

Based on the rigs, I can say that now in enve you can create a rig in the form of steps. (phases)/
Morph, you can probably try creating it using the time remapping function. But of course it would be much more convenient if you could do this with scripts in the frame channel of the shape layer.

**Comment by freewarecamp** [comment]
@AlexKiryanov Thanks. But the rigging capability is not just limited into morphing keyframes. The rig can control any animated property. Thank you for explaining it😁

**Comment by AlexKiryanov** [comment]
Hi all! It's clear.
You can create a slider now. The program initially allows you to do this and then control with this slider any transformation property that the object has.
Sliders in the joystick and DUIK are just shape objects.
The only thing you can’t do now is control morphing. That is, by moving from one frame in the shapes layer to another.
